src/refinement/triquadrisection.jl

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    TriQuadrisection()
+
+Refinement of a mesh by preliminarly triangulating it if needed and
+then subdividing each triangle into four triangles.
+
+## References
+
+* Charles Loop. 1987. [Smooth subdivision surfaces based on 
+  triangles](https://charlesloop.com/thesis.pdf). 
+  Master's thesis, University of Utah.
+"""
+struct TriQuadrisection <: RefinementMethod end
+
+function refine(mesh, ::TriQuadrisection)
+  if !(eltype(mesh) <: Triangle)
+    mesh = simplexify(mesh)
```

Comment:
We shouldn't overwrite user input. I think we need to save the result in a new variable, maybe tmesh